### PR TITLE
test admin dashboard

### PR DESCRIPTION
**Pull request by Floozutter**

### CODE REVIEW

File: baseline/client/tests/dashboard.test.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    {
+        computer: {
+            "browser.name": "Magic",
+            "os.name": "Horn",
+            platform: "Horn",
+            "screen.size": "1920x1080",
+            ua: "Mozilla/5.0 (Horn; rv:97.0) Gecko/20100101 Magic/97.0",
+        },
+        createdAt: "2022-03-01T00:00:00.000Z",
+        email: "bookhorse@canterlot.gov",
```
[the remark this answers is not in the document]
Comment (Floozutter) [reply baseline/client/tests/dashboard.test.js]: Makes sense! I changed them all.

File: baseline/client/tests/dashboard.test.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+        subjectCell,
+        dailyTasksT1Cell,
+        eegT1Cell,
+        mriT1Cell,
+        sessionsCell,
+        eegT2Cell,
+        mriT2Cell,
+        dailyTasksT2Cell,
+    ] = row.querySelectorAll("td");
+    const finishedSets = sets.filter(s => s.experiment === "set-finished").length;
```
[the remark this answers is not in the document]
Comment (Floozutter) [reply baseline/client/tests/dashboard.test.js]: Right! I changed the variables names in `dashboard.js` too.

File: baseline/client/babel.config.js [hunk 3]
```diff
@@ -1,5 +1,9 @@
 module.exports = {
-    plugins: ['@babel/plugin-transform-regenerator', '@babel/plugin-transform-runtime'],
+    plugins: [
+        '@babel/plugin-transform-destructuring',
```
[the remark this answers is not in the document]
Comment (Floozutter) [reply baseline/client/babel.config.js:3]: Oh, this one was a little confusing. Without it, [`npm test -- dashboard` fails](https://youtu.be/s2lOfUbyTgg) with:
```
TypeError: C:\Users\flooz\Desktop\pvs\baseline\client\tests\dashboard.test.js: Property name expected type of string but got null
```

I found [[1]](https://github.com/facebook/regenerator/issues/391#issuecomment-617095947) and [[2]](https://stackoverflow.com/a/62392149) after looking up the error. Just to check, I tried commenting out the destructuring assignments in `dashboard.js` and `dashboard.test.js` and, sure enough, that got rid of this particular error message. So I followed their advice and added `@babel/plugin-transform-destructuring`, which also got rid of the error.

**Review by noah10 — CHANGES_REQUESTED** [review]
Thanks for the changes - they look good. I just noticed that there are no tests for the checkbox functionality, though. Could you please add some tests to confirm that the api client update function is called (with the expected date) when checking an unchecked box and that unchecking a box displays the confirmation and makes the appropriate update if the confirmation is OK'd?

File: baseline/client/tests/dashboard.test.js [hunk 4]
```diff
@@ -279,22 +280,61 @@ describe("dashboard", () => {
...
+        const twiRow = document.querySelector(`[data-user-id="${twiId}"]`);
+        const [_, __, ___, mriT1Cell, ...____] = twiRow.querySelectorAll("td");
+        expect(mc.users.get(twiId).progress?.mriT1).toBeTruthy();
+        expect(mriT1Cell.querySelector("input").checked).toBe(true);
+        expect(mriT1Cell.querySelector("span").textContent).toBeTruthy();
+        // uncheck Twilight Sparkle's MRI T1 checkbox
+        mriT1Cell.querySelector("input").dispatchEvent(new MouseEvent("click", {bubbles: true}));
+        // wait for the async click event handler to resolve
+        await new Promise(process.nextTick);
+        // Twilight Sparkle should now have a timestamp for MRI T1
```
Comment (noah10) [inline baseline/client/tests/dashboard.test.js:332]: I think "now" is probably supposed to be "not" here.

**Review by noah10 — CHANGES_REQUESTED** [review]
Other than the one little issue with the comment that says the opposite of what it should, I think this is good to go.

File: baseline/client/tests/dashboard.test.js (hunk 4 above)
Comment (Floozutter) [reply baseline/client/tests/dashboard.test.js]: Nice catch! Just fixed it.

**Review by noah10 — APPROVED** [review]

**Pull request merged**